### PR TITLE
hwmv2: port qemu-xtensa and xt-sim boards

### PR DESCRIPTION
**Pull request by nordicjm**
tests: Update board names for hwmv2
    
    Updates tests that use board names which have changed with boards v2

    boards: xtensa: qemu_xtensa: Convert to v2
    
    Converts the board to hwmv2
    

    soc: dc233c: Port to HWMv2
    
    Ports the dc233c SoC configuration to hardware model version 2
    

    boards: xtensa: xt-sim: Convert to v2
    
    Converts the board to hwmv2

    soc: xtensa_sample_controller: Port to HWMv2
    
    Ports the xtensa_sample_controller SoC configuration to hardware
    model version 2

### CODE REVIEW

File: boards/v2/qemu/qemu_xtensa/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2017, 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_QEMU_XTENSA || BOARD_QEMU_XTENSA_DC233C_MMU
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_xtensa/Kconfig.defconfig:4]: `BOARD_QEMU_XTENSA_DC233C_MMU` is redundant in this case.
```suggestion
if BOARD_QEMU_XTENSA
```

File: boards/v2/qemu/qemu_xtensa/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+config BUILD_OUTPUT_BIN
+	default n
+
+config IPM_CONSOLE_STACK_SIZE
+	default 2048 if IPM_CONSOLE_RECEIVER
+
+endif # BOARD_QEMU_XTENSA || BOARD_QEMU_XTENSA_DC233C_MMU
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_xtensa/Kconfig.defconfig:12]:
```suggestion
endif # BOARD_QEMU_XTENSA
```

File: boards/v2/qemu/qemu_xtensa/Kconfig.qemu_xtensa [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# XTENSA board configuration
+
+# Copyright (c) 2017, 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_QEMU_XTENSA
+	bool
+
+config BOARD_QEMU_XTENSA_DC233C_MMU
+	bool
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_xtensa/Kconfig.qemu_xtensa:10]: Not needed, this is handled here: 
https://github.com/zephyrproject-rtos/zephyr/blob/051c62a76a7d54f04893f6d73f77c1f946493d04/boards/Kconfig.v2#L8
https://github.com/zephyrproject-rtos/zephyr/blob/051c62a76a7d54f04893f6d73f77c1f946493d04/boards/Kconfig.v2#L13

```suggestion
```

File: boards/v2/qemu/qemu_xtensa/board.cmake [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 set(SUPPORTED_EMU_PLATFORMS qemu)
 
-if(CONFIG_BOARD_QEMU_XTENSA OR CONFIG_BOARD_QEMU_XTENSA_MMU)
+if(CONFIG_BOARD_QEMU_XTENSA OR CONFIG_BOARD_QEMU_XTENSA_DC233C_MMU)
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_xtensa/board.cmake:5]: `CONFIG_BOARD_QEMU_XTENSA_DC233C_MMU` is redundant when first check is board without identifier.
```suggestion
if(CONFIG_BOARD_QEMU_XTENSA)
```

**Review by tejlmand — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
that PR has more than just qemu-xtensa... difficult to review...

**Comment by nordicjm** [comment]
> that PR has more than just qemu-xtensa... difficult to review...

Rebased

File: boards/v2/qemu/qemu_xtensa/board.cmake (hunk 4 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_xtensa/board.cmake]: Fixed

File: boards/v2/qemu/qemu_xtensa/Kconfig.qemu_xtensa (hunk 3 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_xtensa/Kconfig.qemu_xtensa]: Fixed

File: boards/v2/qemu/qemu_xtensa/Kconfig.defconfig (hunk 2 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_xtensa/Kconfig.defconfig]: Fixed

File: boards/v2/qemu/qemu_xtensa/Kconfig.defconfig (hunk 1 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_xtensa/Kconfig.defconfig]: Fixed

**Review by tejlmand — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Comment by tejlmand** [comment]
@nordicjm just discovered this PR lacks a `boards/v2/qemu/qemu_xtensa/Kconfig.qemu_xtensa`

**Comment by tejlmand** [comment]
Followup PR: https://github.com/zephyrproject-rtos/zephyr/pull/68323

**Pull request merged**